### ISSUE TITLE
Redis::connect(): Argument #2 ($port) must be of type int, string given

### ISSUE THREAD

**Issue by mscott83** [laravel/framework#36693]
- Laravel Version: 8.32.1
- PHP Version: 8.0
- Database Driver & Version: Redis 6.2

### Description:
I just switched from predis to phpredis. In my Sail dev environment I have no issues at all, but having built my app into an Alpine-based container before deploying to a Kubernetes cluster, I get the following error whenever I trigger a database query:

[2021-03-21 21:11:01] laravel.ERROR: Redis::connect(): Argument #2 ($port) must be of type int, string given {"exception":"[object] (TypeError(code: 0): Redis::connect(): Argument #2 ($port) must be of type int, string given at /var/www/vendor/laravel/framework/src/Illuminate/Redis/Connectors/PhpRedisConnector.php:141)
 
### Steps To Reproduce:
My .env file looks like this:
REDIS_HOST=redis
REDIS_PORT=6379

and config/database.php looks like this:
    'redis' => [

        'client' => 'phpredis',

        'default' => [
            'host' => env('REDIS_HOST', '127.0.0.1'),
            'password' => env('REDIS_PASSWORD', null),
            'port' => env('REDIS_PORT', 6379),
            'database' => 0,
        ],

        'cache' => [
            'host' => env('REDIS_HOST', '127.0.0.1'),
            'password' => env('REDIS_PASSWORD', null),
            'port' => env('REDIS_PORT', 6379),
            'database' => env('REDIS_CACHE_DB', 1),
        ],
    ],

If I comment out the REDIS_PORT value and just rely on Laravel's default values then I get the exact same error so I don't think it's anything to do with the way I'm formatting my config files.

**Comment by andrew-nuwber** [comment]
Hi, did you solve the issue?

**Comment by mscott83** [comment]
Not solved, but I found a workaround. I was using a ConfigMap in k8s to define my .env file and the Redis port was listed in that as REDIS_PORT=6379. In this configuration I received the error.

As a test I removed the value from the .env ConfigMap and instead specified it as an enviornment variable on the pod instead like so:
            - name: REDIS_PORT
              value: '6379'
and that was enough to get it working.

At this point I'm not sure if this is a Laravel bug, a ConfigMap bug, or just something peculiar to my app, so I thought i'd close the issue for now until I can do a bit more investigation. I'm leaning towards it being a Laravel bug because leaving the value blank is enough to trigger the issue, but I haven't been able to replicate this in my dev environment.

**Comment by andrew-nuwber** [comment]
I just fixed it on my CI environment by adding cast to int

```php
    'redis' => [
        'client' => 'phpredis',

        'default' => [
            'host' => env('REDIS_HOST', '127.0.0.1'),
            'password' => env('REDIS_PASSWORD', null),
            'port' => (int) env('REDIS_PORT', 6379),
            'database' => env('REDIS_DB', 0),
        ],

        'cache' => [
            'host' => env('REDIS_HOST', '127.0.0.1'),
            'password' => env('REDIS_PASSWORD', null),
            'port' => (int) env('REDIS_PORT', 6379),
            'database' => env('REDIS_CACHE_DB', 1),
        ],
    ],
```

**Comment by mfn** [comment]
@mscott83 I had a similar issue recently and it was related to the k8s default behaviour to populate env vars from all services.

Here's a good issue showing this from https://github.com/kubernetes/kubernetes/issues/60099
> `KUBERNETES_PORT=tcp://10.233.0.1:443`

If your service is called `redis`, you will end up with `REDIS_PORT=tcp://…` and this will break the apps, naturally.

No amount of casting in the configs can fix this, your fix by providing an explicit env var is correct.

---

There is, however, another angle to this: the whole feature can be disabled. Because in my case I had _multiple_ applications not working correctly due to this.

TL;DR: I ended up setting `enableServiceLinks: false` as per https://kubernetes.io/docs/concepts/services-networking/service/#environment-variables
> Note: If the service environment variables are not desired (because possible clashing with expected program ones, too many variables to process, only using DNS, etc) you can disable this mode by setting the enableServiceLinks flag to false on the pod spec.

After that you can remove your added env var and it will work again.

HTH
